conf/standalone.conf

```diff
@@ -246,6 +246,19 @@ managedLedgerCursorMaxEntriesPerLedger=50000
 # Max time before triggering a rollover on a cursor ledger
 managedLedgerCursorRolloverTimeInSeconds=14400
 
+# Max number of "acknowledgment holes" that are going to be persistently stored.
+# When acknowledging out of order, a consumer will leave holes that are supposed
+# to be quickly filled by acking all the messages. The information of which
+# messages are acknowledged is persisted by compressing in "ranges" of messages
+# that were acknowledged. After the max number of ranges is reached, the information
+# will only be tracked in memory and messages will be redelivered in case of
+# crashes.
+managedLedgerMaxUnackedRangesToPersist=1000
+
+# Max number of "acknowledgment holes" that can be stored in Zookeeper. If number of unack message range is higher
+# than this limit then broker will persist unacked ranges into bookkeeper to avoid additional data overhead into
+# zookeeper.  
+managedLedgerMaxUnackedRangesToPersistInZooKeeper=1000
```

Comment:
I don't see a lot of value to have this being configurable. It can be confusing to properly understand the scope and meaning of this variable.

Writing the last acked position in the z-node during topic close is just an optimization. I think we could use a constant for the threshold, to a value that has just minimal overhead in the z-node payload. (I'd take it down to 100 entries even)

Reply:
Umm.. The reason I made it configurable because this number really depends on the scale of the system. One can just store 1K-10K ranges in zk if zk footprint size is not huge. But in case of large zk-footprint size we want to avoid storing large unack-range (As per #742 test: 100 msg can take 1KB (existing size is = 80 bytes)) so, in that case we want to keep this value = 10 instead of 100. So, it seems this value may vary based on zk-size..??

Reply:
I think having this initially would be useful for us to tune and realize the suitable size. Once we are comfortable with a value, we can remove the config.

Reply:
@merlimat any thoughts? should we remove it or we can keep default value=10?

Reply:
My vote would be not to add this config option since it seems a bit confusing for user. If the consensus is in the other direction I can live with that. 

Also, please raise the other limit (`managedLedgerMaxUnackedRangesToPersist`) to 10K

Reply:
> Also, please raise the other limit (managedLedgerMaxUnackedRangesToPersist) to 10K

Sure, updated.